### PR TITLE
Basic accessibility

### LINKED ISSUE #46: Tab through custom search suggestions
It would be nice, especially for accessibility, if you could tab through the search suggestions.

### PR DESCRIPTION
**Pull request by LiamBrandt**
closes #46, addresses #48

I used the accessibility tree inspector to check if all the roles were assigned in a reasonable way. Tabbing through buttons and inputs is more obvious now.

### CODE REVIEW

File: app/events.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+/*
+  If space or enter is pressed while focused on an element then click that
+  element.
+*/
+function clickOnSpaceOrEnter(event: KeyboardEvent) {
```
Comment (Anbranin) [inline app/events.tsx]: I'm surprised there isn't more built-in accessibility features like this

File: app/events.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    event.keyCode === 32 || event.keyCode === 13
+  ) {
+    _clickElement(event.target as HTMLElement);
+  }
+}
+
+// Send a mouse click event to an element
+function _clickElement(element: HTMLElement) {
+  const event = document.createEvent('MouseEvent');
+  event.initMouseEvent('click', true, true, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);
```
Comment (Anbranin) [inline app/events.tsx]: The arguments to this method are unreadable. 5 `0`s in a row followed by 4 `false`s without an explanation of what they mean is so uncool man

Comment (LiamBrandt) [reply app/events.tsx]: Ill leave a comment here. I was using a much more obvious way to create a mouse event, but this is the only way to do it in IE, with unnamed countless parameters.

File: app/events.tsx (hunk 1 above)
Comment (LiamBrandt) [reply app/events.tsx]: I think its default built into `button` elements, but not `div`s like we are using. Esri also uses keydown events for these things on their buttons I think.

**Pull request merged**